test/unit/CMakeLists.txt

```diff
@@ -1,10 +1,40 @@
+# Build unit test mechanisms.
+
+set(test_mechanisms
+    celsius_test
```

Comment:
Let's make a ticket for moving the other test-only mechanisms (e.g. `test_kin1` etc) into here.

Reply:
Also a more detailed comment above that makes it clear that these are mechanisms used only in unit tests, hence why we don't make them elsewhere.

Reply:
Updated comment in-place.